File: backend/src/main/kotlin/io/github/projektalmanac/comunidades/entity/Comunidad.java
package io.github.projektalmanac.comunidades.entity;

import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Comunidad {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String nombre;
    private String descripccion;
    private String categoria;
    private String tipo;
    @ManyToMany
    @JoinTable(
            name = "Usuarios",
            joinColumns = @JoinColumn(name = "user_id"),
            inverseJoinColumns = @JoinColumn(name = "comunidad_id")
    )
    private List<User> users = new ArrayList<>();

}

File: backend/src/main/kotlin/io/github/projektalmanac/comunidades/entity/Comunidad.kt
package io.github.projektalmanac.comunidades.entity

import jakarta.persistence.*

@Entity
class Comunidad {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
    var nombre: String? = null
    var descripcion: String? = null

    @ManyToOne(cascade = [CascadeType.ALL])
    var dueno: User? = null

    var visibilidad: String? = null
    var categoria: String? = null
    var descripcionCorta: String? = null
    
    @ManyToMany(cascade = [CascadeType.ALL])
    @JoinTable(
        name = "Usuarios",
        joinColumns = [JoinColumn(name = "user_id")],
        inverseJoinColumns = [JoinColumn(name = "comunidad_id")]
    )

    var usuariosInscritos: MutableList<User> = ArrayList()
}
